### ISSUE TITLE
Sky shader renders artifacts on NVIDIA cards

### ISSUE THREAD

**Issue by yaelatletl** [godotengine/godot#75771]
### Godot version

4.0.2 stable

### System information

Windows 11, RX 5700. Windows 10, RTX 3050. 

### Issue description

The sky shader renders differently on AMD and NVIDIA cards you can note the difference here: 
Windows 11, AMD RX 5700: 
![image](https://user-images.githubusercontent.com/28612272/230539231-7506f1e0-f6c6-4bcd-bf06-a9e8b5f1f66c.png)
Windows 10, RTX 3050: 
(If you see this message, refresh the page in a minute, I'm uploading the screenshot from the other PC) 

### Steps to reproduce

Just open the Sky.tscn in there, you will notice strange artifacts on the ground and no clouds. 

### Minimal reproduction project

[Godot_4_sky_shader-main.zip](https://github.com/godotengine/godot/files/11175961/Godot_4_sky_shader-main.zip)

**Comment by and-rad** [comment]
So this seems to be related to the precision modifiers in the shader. A simple search&replace for `lowp` fixes the artifacts and the disappearing clouds on nVidia (RTX 2060, Win 10). Narrowing it down, the main offenders are `clouds_3d()` and `SphereIntersect()` for the clouds and the noise functions for the artifacts.

I'm not experienced enough with GPU stuff to guess if this is expected, but I found this in the Godot docs

> Some architectures (mainly mobile) can benefit significantly from this **[lower precision]**, but there are downsides such as the additional overhead of conversion between precisions. Refer to the documentation of the target architecture for further information. In many cases, mobile drivers cause inconsistent or unexpected behavior and it is best to avoid specifying precision unless necessary.

and on https://www.khronos.org/opengl/wiki/Type_Qualifier_(GLSL) I found this passage

> Note: Precision qualifiers in GLSL are supported for compatibility with [OpenGL ES](https://www.khronos.org/opengl/wiki/OpenGL_ES). They use the same syntax as ES's qualifiers, but they have no functional effects. Do not use them unless you want your shaders to be ES compatible.

which sounds to me that using lower precision might not have an actual performance gain on desktop systems?

**Comment by clayjohn** [comment]
> which sounds to me that using lower precision might not have an actual performance gain on desktop systems?

That's right. For the most part desktop drivers ignore precision qualifiers entirely. But some drivers do respect them, which is likely what leads to the issue on that specific NVidia card

**Comment by and-rad** [comment]
Wait, am I getting this right then, that it appears wrong on the nVidia GPU because that one renders the shader *more* correctly than the AMD card?

So how should this be treated then? Is it a driver bug? It sounds like a driver bug... well, "bug". And the documentation already advises against using precision modifiers in most cases. Should the shader compiler strip `lowp`, `highp` and `mediump` away when translating to GLSL on desktop platforms?

**Comment by clayjohn** [comment]
Its not a driver bug. The precision hints are just hints that essentially set a lower bound. The drivers are always allowed to use a higher level of precision if they want. 

I'm closing as this is a bug in the user's code, not in the engine
